File: DotNet11/StatusBarTextTestVB/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Imports EWSoftware.StatusBarText

Public Class Form1
    Inherits System.Windows.Forms.Form

    Friend WithEvents sbMessage As EWSoftware.StatusBarText.StatusBarTextProvider
    Friend WithEvents mainMenu1 As System.Windows.Forms.MainMenu
    Friend WithEvents menuItem1 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem2 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem3 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem4 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem5 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem6 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem7 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem8 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem9 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem10 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem11 As System.Windows.Forms.MenuItem
    Friend WithEvents sbStatusBar As System.Windows.Forms.StatusBar
    Friend WithEvents sbTextPanel As System.Windows.Forms.StatusBarPanel
    Friend WithEvents statusBarPanel1 As System.Windows.Forms.StatusBarPanel
    Friend WithEvents gripSpacer As System.Windows.Forms.StatusBarPanel
    Friend WithEvents statusBarPanel2 As System.Windows.Forms.StatusBarPanel
    Friend WithEvents menuItem12 As System.Windows.Forms.MenuItem
    Friend WithEvents menuItem13 As System.Windows.Forms.MenuItem

#Region " Windows Form Designer generated code "

    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Tell the StatusBarTextProvider component the status bar to use
        StatusBarTextProvider.ApplicationStatusBar = sbStatusBar

        ' You can also use any panel you like.  The default is zero
        ' (the left-most panel).  For the demo, we'll use the one
        ' in the middle.
        StatusBarTextProvider.ApplicationDisplayPanel = 1
    End Sub

    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container
        Me.sbMessage = New EWSoftware.StatusBarText.StatusBarTextProvider(Me.components)
        Me.menuItem2 = New System.Windows.Forms.MenuItem
        Me.menuItem3 = New System.Windows.Forms.MenuItem
        Me.menuItem4 = New System.Windows.Forms.MenuItem
        Me.menuItem6 = New System.Windows.Forms.MenuItem
        Me.menuItem7 = New System.Windows.Forms.MenuItem
        Me.menuItem8 = New System.Windows.Forms.MenuItem
        Me.menuItem9 = New System.Windows.Forms.MenuItem
        Me.menuItem11 = New System.Windows.Forms.MenuItem
        Me.menuItem13 = New System.Windows.Forms.MenuItem
        Me.menuItem1 = New System.Windows.Forms.MenuItem
        Me.menuItem5 = New System.Windows.Forms.MenuItem
        Me.menuItem10 = New System.Windows.Forms.MenuItem
        Me.menuItem12 = New System.Windows.Forms.MenuItem
        Me.mainMenu1 = New System.Windows.Forms.MainMenu
        Me.sbStatusBar = New System.Windows.Forms.StatusBar
        Me.statusBarPanel2 = New System.Windows.Forms.StatusBarPanel
        Me.sbTextPanel = New System.Windows.Forms.StatusBarPanel
        Me.statusBarPanel1 = New System.Windows.Forms.StatusBarPanel
        Me.gripSpacer = New System.Windows.Forms.StatusBarPanel
        CType(Me.statusBarPanel2, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.sbTextPanel, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.statusBarPanel1, System.ComponentModel.ISupportInitialize).BeginInit()
        CType(Me.gripSpacer, System.ComponentModel.ISupportInitialize).BeginInit()
        Me.SuspendLayout()
        '
        'menuItem2
        '
        Me.menuItem2.Index = 0
        Me.sbMessage.SetStatusBarText(Me.menuItem2, "Load a file of some sort")
        Me.menuItem2.Text = "&Load"
        '
        'menuItem3
        '
        Me.menuItem3.Index = 1
        Me.sbMessage.SetStatusBarText(Me.menuItem3, "Save a file of some sort")
        Me.menuItem3.Text = "&Save"
        '
        'menuItem4
        '
        Me.menuItem4.Index = 2
        Me.sbMessage.SetStatusBarText(Me.menuItem4, "Exit the application")
        Me.menuItem4.Text = "&Exit"
        '
        'menuItem6
        '
        Me.menuItem6.Index = 0
        Me.sbMessage.SetStatusBarText(Me.menuItem6, "Open a new window")
        Me.menuItem6.Text = "&New"
        '
        'menuItem7
        '
        Me.menuItem7.Index = 1
        Me.sbMessage.SetStatusBarText(Me.menuItem7, "View the window list")
        Me.menuItem7.Text = "&List"
        '
        'menuItem8
        '
        Me.menuItem8.Index = 2
        Me.sbMessage.SetStatusBarText(Me.menuItem8, "Minimize the window")
        Me.menuItem8.Text = "&Minimize"
        '
        'menuItem9
        '
        Me.menuItem9.Index = 3
        Me.sbMessage.SetStatusBarText(Me.menuItem9, "Maximize the window")
        Me.menuItem9.Text = "M&aximize"
        '
        'menuItem11
        '
        Me.menuItem11.Index = 0
        Me.sbMessage.SetStatusBarText(Me.menuItem11, "View copyright and contact information")
        Me.menuItem11.Text = "&About"
        '
        'menuItem13
        '
        Me.menuItem13.Index = 2
        Me.sbMessage.SetShowAsBlank(Me.menuItem13, True)
        Me.menuItem13.Text = "&Blank status text on this one"
        '
        'menuItem1
        '
        Me.menuItem1.Index = 0
        Me.menuItem1.MenuItems.AddRange(New System.Windows.Forms.MenuItem() {Me.menuItem2, Me.menuItem3, Me.menuItem4})
        Me.sbMessage.SetStatusBarText(Me.menuItem1, "File related commands")
        Me.menuItem1.Text = "&File"
        '
        'menuItem5
        '
        Me.menuItem5.Index = 1
        Me.menuItem5.MenuItems.AddRange(New System.Windows.Forms.MenuItem() {Me.menuItem6, Me.menuItem7, Me.menuItem8, Me.menuItem9})
        Me.sbMessage.SetStatusBarText(Me.menuItem5, "Window related commands")
        Me.menuItem5.Text = "&Window"
        '
        'menuItem10
        '
        Me.menuItem10.Index = 2
        Me.menuItem10.MenuItems.AddRange(New System.Windows.Forms.MenuItem() {Me.menuItem11, Me.menuItem12, Me.menuItem13})
        Me.sbMessage.SetShowAsBlank(Me.menuItem10, True)
        Me.menuItem10.Text = "&Help"
        '
        'menuItem12
        '
        Me.menuItem12.Index = 1
        Me.menuItem12.Text = "-"
        '
        'mainMenu1
        '
        Me.mainMenu1.MenuItems.AddRange(New System.Windows.Forms.MenuItem() {Me.menuItem1, Me.menuItem5, Me.menuItem10})
        '
        'sbStatusBar
        '
        Me.sbStatusBar.Location = New System.Drawing.Point(0, 513)
        Me.sbStatusBar.Name = "sbStatusBar"
        Me.sbStatusBar.Panels.AddRange(New System.Windows.Forms.StatusBarPanel() {Me.statusBarPanel2, Me.sbTextPanel, Me.statusBarPanel1, Me.gripSpacer})
        Me.sbStatusBar.ShowPanels = True
        Me.sbStatusBar.Size = New System.Drawing.Size(816, 22)
        Me.sbStatusBar.TabIndex = 0
        '
        'statusBarPanel2
        '
        Me.statusBarPanel2.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Contents
        Me.statusBarPanel2.Text = "Test"
        Me.statusBarPanel2.Width = 41
        '
        'sbTextPanel
        '
        Me.sbTextPanel.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Spring
        Me.sbTextPanel.Text = "Ready"
        Me.sbTextPanel.Width = 708
        '
        'statusBarPanel1
        '
        Me.statusBarPanel1.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Contents
        Me.statusBarPanel1.Text = "Test"
        Me.statusBarPanel1.Width = 41
        '
        'gripSpacer
        '
        Me.gripSpacer.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Contents
        Me.gripSpacer.BorderStyle = System.Windows.Forms.StatusBarPanelBorderStyle.None
        Me.gripSpacer.Width = 10
        '
        'Form1
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(6, 15)
        Me.ClientSize = New System.Drawing.Size(816, 535)
        Me.Controls.Add(Me.sbStatusBar)
        Me.IsMdiContainer = True
        Me.Menu = Me.mainMenu1
        Me.Name = "Form1"
        Me.Text = "StatusBarTextProvider Demo"
        CType(Me.statusBarPanel2, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.sbTextPanel, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.statusBarPanel1, System.ComponentModel.ISupportInitialize).EndInit()
        CType(Me.gripSpacer, System.ComponentModel.ISupportInitialize).EndInit()
        Me.ResumeLayout(False)

    End Sub

#End Region

    ' Show the test forms with a bunch of controls as an MDI child
    Private Sub menuItem2_Click(sender As Object, e As System.EventArgs) _
      Handles menuItem2.Click
        Dim mdiForm As New Form2()
        mdiForm.MdiParent = Me
        mdiForm.Show()
    End Sub

    ' Show the test forms with a bunch of controls as a modal dialog
    Private Sub menuItem11_Click(sender As Object, e As System.EventArgs) _
      Handles menuItem11.Click
        Dim dlg As New Form2()
        dlg.ShowDialog()
        dlg.Dispose()
    End Sub

    ' Exit the application
    Private Sub menuItem4_Click(sender As Object, e As System.EventArgs) _
      Handles menuItem4.Click
        Me.Close()
    End Sub

End Class

File: DotNet11/StatusBarTextTestCS/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;

using EWSoftware.StatusBarText;

namespace StatusBarTextTest
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
        private EWSoftware.StatusBarText.StatusBarTextProvider sbMessage;
        private System.Windows.Forms.MainMenu mainMenu1;
        private System.Windows.Forms.MenuItem menuItem1;
        private System.Windows.Forms.MenuItem menuItem2;
        private System.Windows.Forms.MenuItem menuItem3;
        private System.Windows.Forms.MenuItem menuItem4;
        private System.Windows.Forms.MenuItem menuItem5;
        private System.Windows.Forms.MenuItem menuItem6;
        private System.Windows.Forms.MenuItem menuItem7;
        private System.Windows.Forms.MenuItem menuItem8;
        private System.Windows.Forms.MenuItem menuItem9;
        private System.Windows.Forms.MenuItem menuItem10;
        private System.Windows.Forms.MenuItem menuItem11;
        private System.Windows.Forms.StatusBar sbStatusBar;
        private System.Windows.Forms.StatusBarPanel sbTextPanel;
        private System.Windows.Forms.StatusBarPanel statusBarPanel1;
        private System.Windows.Forms.StatusBarPanel gripSpacer;
        private System.Windows.Forms.StatusBarPanel statusBarPanel2;
        private System.Windows.Forms.MenuItem menuItem12;
        private System.Windows.Forms.MenuItem menuItem13;
        private System.ComponentModel.IContainer components;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

            // Tell the StatusBarTextProvider component the status bar to use
            StatusBarTextProvider.ApplicationStatusBar = sbStatusBar;

            // You can also use any panel you like.  The default is zero
            // (the left-most panel).  For the demo, we'll use the one
            // in the middle.
            StatusBarTextProvider.ApplicationDisplayPanel = 1;
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.components = new System.ComponentModel.Container();
            this.sbMessage = new EWSoftware.StatusBarText.StatusBarTextProvider(this.components);
            this.menuItem2 = new System.Windows.Forms.MenuItem();
            this.menuItem3 = new System.Windows.Forms.MenuItem();
            this.menuItem4 = new System.Windows.Forms.MenuItem();
            this.menuItem6 = new System.Windows.Forms.MenuItem();
            this.menuItem7 = new System.Windows.Forms.MenuItem();
            this.menuItem8 = new System.Windows.Forms.MenuItem();
            this.menuItem9 = new System.Windows.Forms.MenuItem();
            this.menuItem11 = new System.Windows.Forms.MenuItem();
            this.menuItem13 = new System.Windows.Forms.MenuItem();
            this.menuItem1 = new System.Windows.Forms.MenuItem();
            this.menuItem5 = new System.Windows.Forms.MenuItem();
            this.menuItem10 = new System.Windows.Forms.MenuItem();
            this.menuItem12 = new System.Windows.Forms.MenuItem();
            this.mainMenu1 = new System.Windows.Forms.MainMenu();
            this.sbStatusBar = new System.Windows.Forms.StatusBar();
            this.statusBarPanel2 = new System.Windows.Forms.StatusBarPanel();
            this.sbTextPanel = new System.Windows.Forms.StatusBarPanel();
            this.statusBarPanel1 = new System.Windows.Forms.StatusBarPanel();
            this.gripSpacer = new System.Windows.Forms.StatusBarPanel();
            ((System.ComponentModel.ISupportInitialize)(this.statusBarPanel2)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.sbTextPanel)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.statusBarPanel1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.gripSpacer)).BeginInit();
            this.SuspendLayout();
            // 
            // menuItem2
            // 
            this.menuItem2.Index = 0;
            this.sbMessage.SetStatusBarText(this.menuItem2, "Load a file of some sort");
            this.menuItem2.Text = "&Load";
            this.menuItem2.Click += new System.EventHandler(this.menuItem2_Click);
            // 
            // menuItem3
            // 
            this.menuItem3.Index = 1;
            this.sbMessage.SetStatusBarText(this.menuItem3, "Save a file of some sort");
            this.menuItem3.Text = "&Save";
            // 
            // menuItem4
            // 
            this.menuItem4.Index = 2;
            this.sbMessage.SetStatusBarText(this.menuItem4, "Exit the application");
            this.menuItem4.Text = "&Exit";
            this.menuItem4.Click += new System.EventHandler(this.menuItem4_Click);
            // 
            // menuItem6
            // 
            this.menuItem6.Index = 0;
            this.sbMessage.SetStatusBarText(this.menuItem6, "Open a new window");
            this.menuItem6.Text = "&New";
            // 
            // menuItem7
            // 
            this.menuItem7.Index = 1;
            this.sbMessage.SetStatusBarText(this.menuItem7, "View the window list");
            this.menuItem7.Text = "&List";
            // 
            // menuItem8
            // 
            this.menuItem8.Index = 2;
            this.sbMessage.SetStatusBarText(this.menuItem8, "Minimize the window");
            this.menuItem8.Text = "&Minimize";
            // 
            // menuItem9
            // 
            this.menuItem9.Index = 3;
            this.sbMessage.SetStatusBarText(this.menuItem9, "Maximize the window");
            this.menuItem9.Text = "M&aximize";
            // 
            // menuItem11
            // 
            this.menuItem11.Index = 0;
            this.sbMessage.SetStatusBarText(this.menuItem11, "View copyright and contact information");
            this.menuItem11.Text = "&About";
            this.menuItem11.Click += new System.EventHandler(this.menuItem11_Click);
            // 
            // menuItem13
            // 
            this.menuItem13.Index = 2;
            this.sbMessage.SetShowAsBlank(this.menuItem13, true);
            this.menuItem13.Text = "&Blank status text on this one";
            // 
            // menuItem1
            // 
            this.menuItem1.Index = 0;
            this.menuItem1.MenuItems.AddRange(new System.Windows.Forms.MenuItem[] {
                                                                                      this.menuItem2,
                                                                                      this.menuItem3,
                                                                                      this.menuItem4});
            this.sbMessage.SetStatusBarText(this.menuItem1, "File related commands");
            this.menuItem1.Text = "&File";
            // 
            // menuItem5
            // 
            this.menuItem5.Index = 1;
            this.menuItem5.MenuItems.AddRange(new System.Windows.Forms.MenuItem[] {
                                                                                      this.menuItem6,
                                                                                      this.menuItem7,
                                                                                      this.menuItem8,
                                                                                      this.menuItem9});
            this.sbMessage.SetStatusBarText(this.menuItem5, "Window related commands");
            this.menuItem5.Text = "&Window";
            // 
            // menuItem10
            // 
            this.menuItem10.Index = 2;
            this.menuItem10.MenuItems.AddRange(new System.Windows.Forms.MenuItem[] {
                                                                                       this.menuItem11,
                                                                                       this.menuItem12,
                                                                                       this.menuItem13});
            this.sbMessage.SetShowAsBlank(this.menuItem10, true);
            this.menuItem10.Text = "&Help";
            // 
            // menuItem12
            // 
            this.menuItem12.Index = 1;
            this.menuItem12.Text = "-";
            // 
            // mainMenu1
            // 
            this.mainMenu1.MenuItems.AddRange(new System.Windows.Forms.MenuItem[] {
                                                                                      this.menuItem1,
                                                                                      this.menuItem5,
                                                                                      this.menuItem10});
            // 
            // sbStatusBar
            // 
            this.sbStatusBar.Location = new System.Drawing.Point(0, 513);
            this.sbStatusBar.Name = "sbStatusBar";
            this.sbStatusBar.Panels.AddRange(new System.Windows.Forms.StatusBarPanel[] {
                                                                                           this.statusBarPanel2,
                                                                                           this.sbTextPanel,
                                                                                           this.statusBarPanel1,
                                                                                           this.gripSpacer});
            this.sbStatusBar.ShowPanels = true;
            this.sbStatusBar.Size = new System.Drawing.Size(816, 22);
            this.sbStatusBar.TabIndex = 0;
            // 
            // statusBarPanel2
            // 
            this.statusBarPanel2.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Contents;
            this.statusBarPanel2.Text = "Test";
            this.statusBarPanel2.Width = 41;
            // 
            // sbTextPanel
            // 
            this.sbTextPanel.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Spring;
            this.sbTextPanel.Text = "Ready";
            this.sbTextPanel.Width = 708;
            // 
            // statusBarPanel1
            // 
            this.statusBarPanel1.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Contents;
            this.statusBarPanel1.Text = "Test";
            this.statusBarPanel1.Width = 41;
            // 
            // gripSpacer
            // 
            this.gripSpacer.AutoSize = System.Windows.Forms.StatusBarPanelAutoSize.Contents;
            this.gripSpacer.BorderStyle = System.Windows.Forms.StatusBarPanelBorderStyle.None;
            this.gripSpacer.Width = 10;
            // 
            // Form1
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(6, 15);
            this.ClientSize = new System.Drawing.Size(816, 535);
            this.Controls.Add(this.sbStatusBar);
            this.IsMdiContainer = true;
            this.Menu = this.mainMenu1;
            this.Name = "Form1";
            this.Text = "StatusBarTextProvider Demo";
            ((System.ComponentModel.ISupportInitialize)(this.statusBarPanel2)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.sbTextPanel)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.statusBarPanel1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.gripSpacer)).EndInit();
            this.ResumeLayout(false);

        }
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

        /// <summary>
        /// Show the test forms with a bunch of controls as an MDI child
        /// </summary>
        /// <param name="sender">The sender of the event</param>
        /// <param name="e">The event arguments</param>
        private void menuItem2_Click(object sender, System.EventArgs e)
        {
            Form2 mdiForm = new Form2();
            mdiForm.MdiParent = this;
            mdiForm.Show();
        }

        /// <summary>
        /// Show the test forms with a bunch of controls as a modal dialog
        /// </summary>
        /// <param name="sender">The sender of the event</param>
        /// <param name="e">The event arguments</param>
        private void menuItem11_Click(object sender, System.EventArgs e)
        {
            using(Form2 dlg = new Form2())
            {
                dlg.ShowDialog();
            }
        }

        /// <summary>
        /// Exit the application
        /// </summary>
        /// <param name="sender">The sender of the event</param>
        /// <param name="e">The event arguments</param>
        private void menuItem4_Click(object sender, System.EventArgs e)
        {
            this.Close();        
        }
	}
}
